### ISSUE TITLE
math.floating-point .double weird output ?

### ISSUE THREAD

**Issue by jonenst** [factor/factor#2767]
```factor
USE: math.floating-point
1.25 double.
0b0
 01111111111
           0100000000000000000000000000000000000000000000000000
! weird ?
```

was the intention the following ?
```factor
USE: math.floating-point
1.25 double.
0
 01111111111
            0100000000000000000000000000000000000000000000000000
! better ?
```
 
 which is the result of the following diff:
 ```diff
diff --git a/basis/math/floating-point/floating-point.factor b/basis/math/floating-point/floating-point.factor
index a3e6f2d124..c96f2166dc 100644
--- a/basis/math/floating-point/floating-point.factor
+++ b/basis/math/floating-point/floating-point.factor
@@ -31,11 +31,11 @@ IN: math.floating-point
 
 : double. ( double -- )
     double>bits
-    [ (double-sign) .b ]
+    [ (double-sign) >bin print ]
     [ (double-exponent-bits) >bin 11 CHAR: 0 pad-head bl print ]
     [
         (double-mantissa-bits) >bin 52 CHAR: 0 pad-head
-        11 [ bl ] times print
+        12 [ bl ] times print
     ] tri ;
 
 : infinity? ( double -- ? )
```
 
 It seems more logical, there are 1 bit of sign and 11 bits of mantissa, so 11+1 = 12
 Also use `>bin print` on the first line just like the 2 other lines ?

**Comment by erg** [comment]
Looks like a good fix!

**Comment by mrjbq7** [comment]
Fixed in 3ce465869d71f5d572325791456570bf68cd54d4.
